rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -47,10 +47,27 @@ pub(crate) fn run(params: client::GuiParams) -> Result<()> {
         inject_faults,
     } = params;
 
+    // Need to keep this alive so crashes will be handled. Dropping detaches it.
+    let _crash_handler = match client::crash_handling::attach_handler() {
+        Ok(x) => Some(x),
+        Err(error) => {
+            tracing::warn!(?error, "Did not set up crash handler");
+            None
+        }
+    };
+
     // Needed for the deep link server
     let rt = tokio::runtime::Runtime::new()?;
     let _guard = rt.enter();
 
+    // If enabled, segfaults the program after 10 seconds to test crash dumps
```

Comment:
I'm not the world's foremost expert in writing Rust unit tests, but would it make sense to modify the function signature for `run` to include a `segfault: bool` that we can then write a unit test for this?

Just trying to keep test codepaths out of the release codepaths (if it's not too cumbersome)

Reply:
Yeah I'm in favor of those kind of things. We had some discussion about that before with the `inject_faults` boolean that would intentionally slow down file I/O to make sure the GUI doesn't hang. It's hard to test things like this without having some "cheat code" paths to break things on purpose.

We can also test it externally by launching the GUI as a subprocess of a unit test and then sending a kill signal to it, maybe. I'm not sure if that will trigger the crash handler on Windows. 🤔